DPGAnalysis/Skims/interface/TriggerMatchProducer.h

```diff
@@ -139,7 +139,7 @@ void TriggerMatchProducer<object>::produce(edm::Event& event, const edm::EventSe
 
     if (TString(*iHLT).Contains(TRegexp(hltTag_))) {
       triggerInMenu[*iHLT] = true;
-      if (hltPrescaleProvider_.prescaleValue(event, eventSetup, *iHLT) == 1)
+      if (hltPrescaleProvider_.prescaleValue<double>(event, eventSetup, *iHLT) == 1)
```

Comment:
Is this ok with rounding effects?

Reply:
To the best of my knowledge, yes, in short because that `double` is not the result of floating-point operation.

This call goes back to [`HLTConfigData::prescaleValue<T>`](https://github.com/cms-sw/cmssw/pull/39220/files#diff-d1e7d2010163f6b4e982bc10451cdc2e50c153e9c8917ebae0a7501bcc480b33R188), where we have the implicit conversion of [`HLTPrescaleTable::prescale`](https://github.com/cms-sw/cmssw/blob/54807ceb71d14ca6d3e9e049ad38ba10f2f0546e/DataFormats/HLTReco/interface/HLTPrescaleTable.h#L37) (unsigned int) into `T` (which must be `double` or `FractionalPrescale`). So, we are reading an `int` as a `double`, not calculating this `double` with a floating-point operation (in that case, the complication of rounding effects would apply, afaiu).

The unit test in [HLTrigger/HLTcore/test/test_catch2_HLTConfigData.cc](https://github.com/cms-sw/cmssw/pull/39220/files/9d3f0c13f596b432b0f056e367c629da76d84b7b#diff-d582cefa382940b57bb0fb63c37da0e5458472b27f20409b088998884d5cce09R126) was extended exactly to gain some confidence in this respect, comparing `double`, or `FractionalPrescale`, to `unsigned int` for the values of a dummy PS table.